### ISSUE TITLE
add option to make contaminant stars that pass qso/lrg/elg/bgs cuts

### ISSUE THREAD

**Issue by sbailey** [desihub/desisim#135]
To generate stellar templates as contaminants for non-stellar types, add an option to `STAR.make_templates()` (or `STAR.__init__`?) to specify which target selection cuts should be applied.  Alternatively, this could be done as a separate subclass of `STAR`, similar to `FSTD` and `MWS_STAR`.  

If the target selection cut functions were refactored to have the same arguments for every function, whether the variables were used or not, this functionality could be achieved by passing in the actual function to use for the cuts as an argument to make_templates.

Bottom line is that we need a way to make stars that pass the cuts of other types to use as contaminants for the simulations.  Make it so.

**Comment by sbailey** [comment]
closed via PR #150
